Add Syrian Arab Republic holidays

Wasif-Shahzad opened the pull request:
<!--
  Thanks for contributing to holidays!
-->

## Proposed change

<!--
  Describe the big picture of your changes.
  Don't forget to link your PR to an existing issue if any.
-->

Added Syrian Arab Republic holidays. I was unable to find much detail about the historical data even with deep research assisted by Gemini and GPT-5. There are documents pre-2004 but they don't seem to include any holiday related data so I had to guess the `start_year` part as of now. The same applies for Christmas and Boxing day. They were given off in 2024 but there isn't much of a data about future so Gemini suggested to assume that it'll be a recurring regular public holiday now. 

Fixes #1275 

## Type of change

<!--
  Type of change you want to introduce. Please, check one (1) box only!
  If your PR requires multiple boxes to be checked, most likely it needs to
  be split into multiple PRs.
-->

- [x] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new `holidays` functionality in general)

## Checklist

<!--
  Put an `x` in the boxes that apply. You can change them after PR is created.
-->

- [x] I've read and followed the [contributing guidelines](https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.md).
- [x] I've run `make check` locally; all checks and tests passed.

<!--
  Thanks again for your contribution!
-->

Wasif-Shahzad committed:
Initial commit

Wasif-Shahzad committed:
Merge remote-tracking branch 'upstream/dev' into syria-holidays

Wasif-Shahzad committed:
Update country count

KJhellico commented on holidays/countries/syrian_arab_republic.py:
```diff
@@ -0,0 +1,166 @@
+class SyriaIslamicHolidays(_CustomIslamicHolidays):
+    # https://web.archive.org/web/20250120140034/https://www.timeanddate.com/holidays/syria/muharram-new-year
+    HIJRI_NEW_YEAR_DATES_CONFIRMED_YEARS = (2020, 2025)
+    HIJRI_NEW_YEAR_DATES = {
+        2020: (AUG, 20),
+        2021: (AUG, 9),
+        2022: (JUL, 30),
+        2023: (JUL, 19),
+        2024: (JUL, 7),
+        2025: (JUN, 26),
+    }
```
```suggestion
```
Leave only the dates that differ from the standard ones.

KJhellico commented on holidays/countries/syrian_arab_republic.py:
```diff
@@ -0,0 +1,166 @@
+    # https://web.archive.org/web/20240808150300/https://www.timeanddate.com/holidays/syria/prophet-birthday
+    MAWLID_DATES_CONFIRMED_YEARS = (2020, 2024)
+    MAWLID_DATES = {
+        2020: (OCT, 29),
+        2021: (OCT, 19),
+        2022: (OCT, 8),
+        2023: (SEP, 27),
+        2024: (SEP, 15),
```
```suggestion
        2021: (OCT, 19),
```

KJhellico commented on holidays/countries/syrian_arab_republic.py:
```diff
@@ -0,0 +1,166 @@
+
+    # https://web.archive.org/web/20250126132002/https://www.timeanddate.com/holidays/syria/eid-al-adha
+    EID_AL_ADHA_DATES_CONFIRMED_YEARS = (2020, 2025)
+    EID_AL_ADHA_DATES = {
+        2020: (JUL, 31),
+        2021: (JUL, 20),
+        2022: (JUL, 9),
+        2023: (JUN, 28),
+        2024: (JUN, 16),
+        2025: (JUN, 6),
+    }
```
```suggestion
```

PPsyrius commented on holidays/countries/bhutan.py:
```diff
@@ -21,7 +21,7 @@ class Bhutan(HolidayBase, HinduCalendarHolidays, TibetanCalendarHolidays):
     References:
         * <https://en.wikipedia.org/wiki/Public_holidays_in_Bhutan>
-        * [The Bhutanese calendar](http://www.kalacakra.org/calendar/bhutlist.htm)
+        * [The Bhutanese calendar](https://web.archive.org/web/20250707105447/http://www.kalacakra.org/calendar/bhutlist.htm)
```
Good catch, although IMO this should be added in a separate PR 👍

Wasif-Shahzad committed:
Remove standard dates for Islamic holidays from SyriaIslamicHolidays class

Wasif-Shahzad committed:
Merge remote-tracking branch 'upstream/dev' into syria-holidays

KJhellico commented on holidays/countries/syrian_arab_republic.py:
```diff
@@ -0,0 +1,129 @@
+        # Tishreen Liberation War Day.
+        self._add_holiday_oct_6(tr("حرب تشرين التحريرية"))
+
+        # Revolution Day.
+        self._add_holiday_dec_8(tr("الثورة السورية"))
+
```
@coderabbitai, what sources mention Revolution Day on March 8?

KJhellico replied:
So, it seems that March 8 is really Revolution Day (in commemoration of the 1963 revolution), and December 8 is Fall of Assad Regime Day since 2024.

KJhellico replied:
OK, let's move Revolution Day to March 8 (I don't know sufficiently about the political situation in Syria to understand whether it is relevant under the current government), and not add December 8 for now.

Wasif-Shahzad committed:
Remove archived link from bhutan.py

Wasif-Shahzad committed:
Merge remote-tracking branch 'upstream/dev' into syria-holidays

Wasif-Shahzad committed:
Update names, add Central Bank of Syria reference for 2024

Wasif-Shahzad commented:
I've updated the names based off Central Bank of Syria's Page and added its hyperlink too. I'm waiting for kind of a more confirm verdict of @KJhellico over Revolution Day to implement that.

PPsyrius commented on holidays/countries/syrian_arab_republic.py:
```diff
@@ -0,0 +1,131 @@
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_new_years_day(tr("ليلة رأس السنة الميلادية"))
+
```
@Wasif-Shahzad Let's change it to "New Year's Day" as that's what other sources I've dug up says

- [Library of Congress, Syria Country Profile - 2005](http://memory.loc.gov/frd/cs/profiles/Syria.pdf)
- [Technical Information Center of Iran - 2016](https://web.archive.org/web/20160604125235/http://www.tici.ir/default.aspx?dir=BizInfo_wvn&file=Syria.htm)
- [Daraa Governorate - 2012](https://web.archive.org/web/20120209230102/http://www.daraa-gov.org/modules.php?op=modload&name=Subjects&file=index&req=viewpage&pageid=11)

Wasif-Shahzad committed:
Change New Year's Day's Arabic name

Wasif-Shahzad committed:
Reformat, correct year in setUpClass and add references in SyriaIslamicHolidays

Wasif-Shahzad commented on holidays/locale/ar/LC_MESSAGES/SY.po:
```diff
@@ -0,0 +1,92 @@
+msgid "عيد الأم"
+msgstr ""
+
+#. Gregorian Easter Sunday.
+msgid "عيد الفصح حسب التقويم الغربي"
+msgstr ""
+
+#. Julian Easter Sunday.
+msgid "عيد الفصح حسب التقويم الشرقي"
+msgstr ""
+
```
Again, I went with the arabic version used in Central Bank's page...

Wasif-Shahzad commented on tests/countries/test_syrian_arab_republic.py:
```diff
@@ -0,0 +1,221 @@
+            ("2024-12-26", "يوم الصناديق"),
+        )
+
+    def test_l10n_en_us(self):
+        self.assertLocalizedHolidays(
+            "en_US",
+            ("2024-01-01", "New Year's Day"),
+            ("2024-03-21", "Mother's Day"),
+            ("2024-03-31", "Gregorian Easter Sunday"),
+            ("2024-04-10", "Eid al-Fitr"),
+            ("2024-04-11", "Eid al-Fitr"),
+            ("2024-04-12", "Eid al-Fitr"),
+            ("2024-04-17", "Independence Day"),
+            ("2024-05-01", "Labor Day"),
+            ("2024-05-05", "Julian Easter Sunday"),
+            ("2024-05-06", "Martyrs' Day"),
+            ("2024-06-15", "Eid al-Adha"),
+            ("2024-06-16", "Eid al-Adha"),
+            ("2024-06-17", "Eid al-Adha"),
+            ("2024-06-18", "Eid al-Adha"),
+            ("2024-07-07", "Hijri New Year"),
+            ("2024-09-15", "Prophet's Birthday"),
+            ("2024-10-06", "Tishreen Liberation War Day"),
+            ("2024-12-08", "Revolution Day"),
+            ("2024-12-25", "Christmas Day"),
+            ("2024-12-26", "Boxing Day"),
+        )
```
Both of the revolution day updates aren't supposed to be in this commit. I'm loading them up in the next commit as I saw @KJhellico 's comment after pushing this commit

Wasif-Shahzad committed:
Move Revolution Day to 8th March

Wasif-Shahzad committed:
Update en_US translations for Islamic New Year

KJhellico commented on holidays/countries/syrian_arab_republic.py:
```diff
@@ -0,0 +1,133 @@
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import GREGORIAN_CALENDAR, MAR, OCT
```
```suggestion
from holidays.calendars.gregorian import GREGORIAN_CALENDAR, MAR, OCT, FRI, SAT
```

KJhellico commented on holidays/countries/syrian_arab_republic.py:
```diff
@@ -0,0 +1,133 @@
+    estimated_label = tr("%s (المقدرة)")
+    # Earliest reliable record dates back to 2004.
+    start_year = 2004
+    supported_languages = ("ar", "en_US")
```
```suggestion
    supported_languages = ("ar", "en_US")
    weekend = {FRI, SAT}
```

Wasif-Shahzad committed:
Change weekend to Friday, Saturday

Wasif-Shahzad committed:
Merge remote-tracking branch 'upstream/dev' into syria-holidays

Wasif-Shahzad committed:
Merge remote-tracking branch 'upstream/dev' into syria-holidays

KJhellico commented on holidays/countries/syrian_arab_republic.py:
```diff
@@ -0,0 +1,134 @@
+        # Tishreen Liberation War Day.
+        self._add_holiday_oct_6(tr("ذكرى حرب تشرين التحريرية"))
+
+        if self._year >= 2024:
+            # Christmas Day.
+            self._add_christmas_day(tr("عيد الميلاد"), calendar=GREGORIAN_CALENDAR)
+
+            # Boxing Day.
+            self._add_christmas_day_two(tr("يوم الصناديق"), calendar=GREGORIAN_CALENDAR)
+
```
Such comments are excessive. If there was a known regulatory act that introduced this, it could be specified.

KJhellico reviewed (approved):
LGTM! 👍

PPsyrius reviewed (approved):
LGTM 🇸🇾

arkid15r reviewed (commented):
Thanks for helping with this @Wasif-Shahzad 👍
